kythe/cxx/indexer/cxx/frontend.cc

```diff
@@ -123,11 +123,10 @@ void MaybeNormalizeFileVNames(IndexerJob* job) {
 
 void UpdateJobWdirFromUnit(IndexerJob* job) {
   job->working_directory = job->unit.working_directory();
-  if (!llvm::sys::path::is_absolute(job->working_directory)) {
-    llvm::SmallString<1024> stored_wd;
-    CHECK(!llvm::sys::fs::make_absolute(stored_wd));
-    job->working_directory = stored_wd.str();
-  }
+  CHECK(!job->working_directory.empty())
+      << "Indexer jobs must have their absolute working directory set.";
+  CHECK(llvm::sys::path::is_absolute(job->working_directory))
```

Comment:
Does this not imply the previous condition already?

Reply:
Yes, but I expect custom extractors to forget to populate the field entirely more often than to populate it incorrectly.

Reply:
Well, okay, but then should that case have a different diagnostic?

Reply:
The other problem with this solution is that the indexer will now fail on stored compilations that didn't set this value. Maybe instead we should assume "/" if it's not set, and only issue a diagnostic?